app/models/custom_button.rb

```diff
@@ -100,7 +100,8 @@ def invoke(target, source = nil)
     MiqQueue.put(queue_opts(target, args))
   end
 
-  def publish_event(source, target, args)
+  def publish_event(source, target, args = nil)
+    args ||= resource_action.automate_queue_hash(target, {}, User.current_user)
```

Comment:
Would be useful to test that the method calls `resource_action.automate_queue_hash` when called without an `args` parameter since that is the logic being added.